src/edu/washington/escience/myria/operator/CSVFileScanFragment.java

```diff
@@ -179,23 +179,20 @@ public CSVFileScanFragment(
   @Override
   protected TupleBatch fetchNextReady() throws IOException, DbException {
     long lineNumberBegin = lineNumber;
-    boolean unfinishedEncapsulatedTokenRow = false;
+    boolean truncatedQuotedField = false;
 
     while ((buffer.numTuples() < TupleBatch.BATCH_SIZE) && !flagAsIncomplete) {
       lineNumber++;
       if (parser.isClosed()) {
         break;
       }
 
-      if (unfinishedEncapsulatedTokenRow) {
+      if (truncatedQuotedField) {
         onLastRow = true;
       }
 
-      try {
-        if (!onLastRow) {
-          record = iterator.next();
-        }
-      } catch (Exception e) {
+      if (!onLastRow) {
+        record = iterator.next();
```

Comment:
We shouldn't... The quoted field will blow up the hasNext() first. This will happen once ```record``` is the second to last row. We then set the onLastRow flag.

When we loop back up, the truncated field will be marked as onLastRow and we will not call .next() until we read more bytes.